src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -2,192 +2,223 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
+using System;
+using System.ComponentModel;
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.SqlServer.Dac;
+using System.Reflection;
+using System.Collections.Generic;
 
 namespace Microsoft.SqlTools.ServiceLayer.DacFx.Contracts
 {
     /// <summary>
-    /// Class to define deployment options. 
+    /// Class to define deployment option default value and the description
+    /// </summary>
+    public class DeploymentOptionProperty<T>
+    {
+        public DeploymentOptionProperty(T value, string description = "", string displayName = "")
+        {
+            this.Value = value;
+            this.Description = description;
+            this.DisplayName = displayName;
+        }
+
+        // Default and selected value of the deployment options
+        public T Value { get; set; }
+
+        // Description of the deployment options
+        public string Description { get; set; } = string.Empty;
```

Comment:
is setting this to string.Empty necessary? Doesn't description get set to "" by default in the constructor?

Reply:
Not required! its a duplicated one with no reason to have it, Thanks